controller/src/main/java/io/pravega/controller/server/ControllerService.java

```diff
@@ -384,9 +387,24 @@ public CompletableFuture<CreateStreamStatus> createStream(String scope, String s
             return CompletableFuture.completedFuture(
                     CreateStreamStatus.newBuilder().setStatus(CreateStreamStatus.Status.INVALID_STREAM_NAME).build());
         }
+        try {
+            validateStreamConfig(streamConfig);
+        } catch (IllegalArgumentException e) {
+            log.error(requestId, "Create stream: {} failed due to invalid retention policy {}", stream, streamConfig);
```

Comment:
Can we make message as " No retention policy set for stream" message rather than "failed due to invalid retention policy" make it more explicit message, maybe invalid retention policy confuses because they will not make out what is invalid here.

Reply:
modified